### ISSUE TITLE
Block Replace not setting leaf blocks as persistent.

### ISSUE THREAD

**Issue by Moobien** [ldtteam/Structurize#679]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Are you using the latest Structurize Version?

- [X] I am running the latest **alpha** version of Structurize for my Minecraft version.
I am also running the latest versions of other mods that are part of my problem.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies/Structurize [Wiki](https://wiki.minecolonies.ldtteam.com/) or I asked on [discord](https://discord.minecolonies.com/).

### Minecraft Version

1.20

### Structurize Version

737-beta

### MineColonies Version (if related bug)

n/a

### Related Mods and their Versions

n/a

### Current Behavior

replacing any block with a leaf block is not setting the leaf block as persistent.

### Expected Behavior

well..the opposite of what it is doing. replacing a block with a leaf block should set the leaf block as persistent.

### Reproduction Steps

go to dark oak treehouse dyer/tavern on schem server. replace leaf blocks with bedrock, replace bedrock with leaf block and see they are still not persistent.

### Logs

n/a

### Anything else?

n/a

### Footer

---
## Viewers
* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.
n/a

**Comment by Nightenom** [comment]
This is generally problem of the replacement itself, on low level like now it should be just best-guess swap, but it's missing high level where user can modify blockstate properties like persistent

**Comment by Moobien** [comment]
> This is generally problem of the replacement itself, on low level like now it should be just best-guess swap, but it's missing high level where user can modify blockstate properties like persistent

yeah. but it makes schematic fixes a LOT more difficult when you have 1200 leaf blocks to manually replace across a single building (all 5 levels)

**Comment by Nightenom** [comment]
for simple stable replace you can just use fill/replace command

**Comment by uecasm** [comment]
If you replace already-persistent leaves with another leaf type, it should work.  It's only if you replace leaves to a different block type and then back to leaves that you will lose the persistent flag.  This is the same as other blockstate properties at present.

**Comment by Nightenom** [comment]
Related #717
